src/player.cpp

```diff
@@ -6967,6 +6993,11 @@ bool player::has_charges( const itype_id &it, int quantity,
     if( it == "fire" || it == "apparatus" ) {
         return has_fire( quantity );
     }
+    if( it == "UPS" && has_effect( effect_riding ) && mounted_creature &&
+        mounted_creature.get()->has_flag( MF_RIDEABLE_MECH ) ) {
+        auto mons = mounted_creature.get();
+        return quantity <= mons->battery_item->ammo_remaining();
```

Comment:
This return means all the UPS charges of the player inventory are ignored and only those from the mounted creature are used.

However, in `player::use_charges`, you drain charges from the mounted creature *and* continue draining from the player inventory. This is not consistent.